worktree_bsd.go

```diff
@@ -13,8 +13,8 @@ func init() {
 	fillSystemInfo = func(e *index.Entry, sys any) {
 		if os, ok := sys.(*syscall.Stat_t); ok {
 			e.CreatedAt = time.Unix(os.Atimespec.Unix())
-			e.Dev = uint32(os.Dev)
-			e.Inode = uint32(os.Ino)
+			e.Dev = uint32(os.Dev)   //nolint:gosec // G115: dev number fits in uint32
+			e.Inode = uint32(os.Ino) //nolint:gosec // G115: inode fits in uint32
```

Comment:
Both `os.Dev` and `os.Ino` are OS-dependant values. Some implementations are [uint32](https://github.com/golang/go/blob/b28808d838682b2911698fcb934813b02f39fa69/src/syscall/syscall_linux_mips64x.go#L107-L109), but [others](https://github.com/golang/go/blob/b28808d838682b2911698fcb934813b02f39fa69/src/syscall/fs_wasip1.go#L640-L641) are `uint64`, which makes me think in some cases this may overflow.

That being said, upstream relies solely on the [uint32 counterpart](https://github.com/git/git/blob/68cb7f9e92a5d8e9824f5b52ac3d0a9d8f653dbe/statinfo.h#L19C2-L20) for it. Due to that, I'm leaning towards:
- Keep `index.Entry` types as they are.
- Add check for value and error if > `math.MaxUint32`.